apps/avifdec.c

```diff
@@ -248,8 +247,10 @@ int main(int argc, char * argv[])
                        decoder->image->height);
                 ++currIndex;
             }
-            if (nextImageResult != AVIF_RESULT_NO_IMAGES_REMAINING) {
-                printf("ERROR: Failed to decode frame: %s\n", avifResultToString(nextImageResult));
+            if (result == AVIF_RESULT_NO_IMAGES_REMAINING) {
+                result = AVIF_RESULT_OK;
+            } else {
+                printf("ERROR: Failed to decode frame: %s\n", avifResultToString(result));
```

Comment:
Optional: The error message at line 257 should say "parse", not "decode", because it reports the error of the `avifDecoderParse(decoder)` call at line 219.

Compare with lines 291-293.

Note: It seems that the error messages for --info should be printed to stderr instead of stdout. We can deal with this issue later.

Reply:
> Optional: The error message at line 257 should say "parse", not "decode", because it reports the error of the `avifDecoderParse(decoder)` call at line 219.
> 
> Compare with lines 291-293.
> 

Done.

> Note: It seems that the error messages for --info should be printed to stderr instead of stdout. We can deal with this issue later.

Ack. Will do.

Reply:
> Note: It seems that the error messages for --info should be printed to stderr instead of stdout. We can deal with this issue later.

There were only 2 occurrences of this. So i have fixed this as well in this PR.